### DISCUSSION TITLE
Vignette does not update position with camera move

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#669] [Q&A] [answered]
Hello marco!!! Hope you are well and having a good day!

Have a quick question regarding the vignette object. I use it to display info in the vedo powered visualiser that I've built. It seems that despite the fact I am using the same code in both circumstances, the vignette does not follow the camera in both cases. Below is an image:

![vignettes_cropped](https://user-images.githubusercontent.com/8246894/179365005-bab9a00a-e468-4c7c-89e4-e991603e0ded.png)

As you can see in the image, when I tilt the plots, the vignettes behave differently. The left hand one stays flush with the screen as you move it (which is what I want). The right hand one stays static as you move it. In both cases I use the following code to draw the vignette: 

```
sdtxt = f"Coord: {currt2[0]:.2f}, {currt2[1]:.2f}, {currt2[2]:.2f}\n" \
        f"Point: {np.around(pt[0], decimals=2)}, {np.around(pt[1], decimals=2)}, {np.around(pt[2], decimals=2)}\n" \
        f"Str/Dip/DDir: {strike:.2f}/{dipa:.2f}/{dipd:.2f}"

vig = msh.vignette(sdtxt, point=pt, offset=(0.1,0.1), s=0.02, c='w',
                   font="VictorMono").followCamera()
vig.name=f'vig_{msh.name}'
self.add(vig).at(1).render()
```

The plot on the left is a `Plotter` object, the one on the right is extended from the `FreeHandCutPlotter` object. 

Do you have any idea, why this different vignette behaviour is happening? 

Cheers!
j

**Comment by marcomusy** [comment, marked as answer]
Hi! what if you do instead
```
self.at(1)
vig = msh.vignette(sdtxt, point=pt, offset=(0.1,0.1), s=0.02, c='w', font="VictorMono").followCamera()
self.add(vig)
```

Btw the `add()` already calls `render()` so you don't need it

**Reply by jkissi** [reply]
Tried it and it doesn't work, unfortunately...its a strange behaviour. Do you think it could be because the viewer is extended from the FreeHandCutPlotter? Or is that crazy talk?

j

**Reply by marcomusy** [reply]
it may depend on the fact that there is not yet a camera for renderer 1, but i'm not sure.. it depends very much on how you structured your code.

**Reply by jkissi** [reply]
sorry, what do you mean by "there is not yet a camera for renderer 1"?

**Reply by jkissi** [reply]
marco, can you explain the above statement?

**Reply by marcomusy** [reply]
try:
```python
from vedo import *

plt = Plotter(N=2, bg='bb')

sp = Sphere().wireframe()

# this will not work because camera is ambigous
# vig = sp.vignette("The\nNorth Pole", c='k6').followCamera()

plt.at(0)
plt += Cube()

plt.at(1)
vig = sp.vignette("The\nNorth Pole", c='k6').followCamera()
plt += [sp, vig]

plt.show().interactive().close()
```

**Reply by jkissi** [reply]
hi, 
I tried this but got a bit confused. I tried various combinations of the code being uncommented and commented out (including removing both `plt.at(0)` and `plt.at(1)`)but in all cases the vignette behaved correctly! I also removed the sphere and the cube, so it was just the vignette, but it followed the camera, anyway!

I tested the behaviour by removing the `followCamera()` call, and what I am seeing on the right hand side is exactly the same as if `followCamera()` was not there. Very, very strange, I am not really sure what to try next but can you think of a[nother] reason why I could get bahaviour like of haven't called `.followCamera()`, when I have? I am running the latest vedo on anaconda with python 3.10.4....

j

**Reply by marcomusy** [reply]
Uhm not sure why is that, maybe one other quick thing you can try is to pass a camera directly:
```
plt.at(1)
mymesh.followCamera(cam=plt.camera)
```
